elastic/eui
# [Infra] Upgrade dependencies that cause warnings when installing with React 18

## Summary

closes https://github.com/elastic/eui/issues/7080

## QA

- `yarn build-pack`, then move to the `.tgz` file to somewhere you can easily point to
- `npx create-react-app test-eui-18-app && cd test-eui-18-app`
- Install peer dependencies:
```
yarn add @elastic/datemath @emotion/react @emotion/css moment prop-types typescript@4.5 @types/react @types/react-dom
```
- Open `package.json` and add `@elastic/eui` pointed at the local tgz file, e.g. `"@elastic/eui": "/Users/YOURNAME/Documents/elastic-eui-88.4.1.tgz",`
- Open `src/App.js` and import and render an `EuiButton`
- `yarn`
- [x] Confirm no `warning @elastic/eui` messages while installing EUI (CRA has a few of its own, but shouldn't be related to EUI)
- `yarn start`
- [x] Confirm `EuiButton` renders as expected

### General checklist

- Release checklist
